MULTISITE.md

```diff
@@ -0,0 +1,150 @@
+
+Before proceeding, you’ll first need to disable all plugins. Visit the **Plugins » Installed Plugins** page and deactivate all plugins.
+
+Then, navigate to the **Tools » Network Setup** page to configure your multisite network.
+
+![Create a Network of WordPress Sites page](https://i0.wp.com/wordpress.org/support/files/2018/11/network-create.png?fit=1024%2C743&ssl=1)
+
+Next, you need to tell WordPress what kind of domain structure you will be using for your network’s sites: sub-domains (`site1.upstatement.com`) or sub-directories(`upstatement.com/site`). Currently, Skela only supports subdirectories. Select subdirectories.
+
+Provide a title for your network and make sure that the email address in the Network Admin Email field is correct.
+
+Click the `Install` button to continue.
+
+> NOTE: After selecting your domain structure, you cannot change it. If you wish to change your domain structure, refer to the [Troubleshooting](#-troubleshooting) section to destroy the Docker container and start over.
+
+## 🛠 Step 3: Configure Scripts and Nginx
+
+WordPress will then provide you some code you need to add to your files in order to enable your network.
+
+![Enabling the network page](https://i0.wp.com/wordpress.org/support/files/2018/11/tools-network-created.png?fit=1024%2C742&ssl=1)
+
+You can ignore the instructions in the second part regarding `.htaccess` because WordPress assumes you are using an Apache server. Skela uses [nginx](https://www.nginx.com/), not Apache.
+
+Next, open your Skela project in a code editor.
+
+In `scripts/setup.sh`, paste the following code block **below** the line says `define('WP_DEBUG', true);`
+
+```php
+/* Multisite */
+define('MULTISITE', true);
+define('SUBDOMAIN_INSTALL', false);
+define('DOMAIN_CURRENT_SITE', 'YOURSITENAME.ups.dock');
+define('PATH_CURRENT_SITE', '/');
+define('SITE_ID_CURRENT_SITE', 1);
+define('BLOG_ID_CURRENT_SITE', 1);
+```
+
+Don’t forget to replace **YOURSITENAME** with the name of your project.
+
+Notice that you’re not directly modifying `wp-config.php` in the Docker container this time — instead you're adding this code block to the setup script. This is because this script is run when you set up your docker container (`./bin/install`), which we'll be doing shortly. The code block above will automatically be added to `wp-config.php`.
+
+Next, in `conf/nginx/nginx-site.conf`, locate the line where it says `# Override base location to work with WordPress pretty permalinks`. Right **above** this line, paste the following block:
+
+```conf
+# WordPress Multisite Subdirectory Rules
+# https://wordpress.org/support/article/nginx/#wordpress-multisite-subdirectory-rules
+if (!-e $request_filename) {
+  rewrite /wp-admin$ $scheme://$host$request_uri/ permanent;
+  rewrite ^(/[^/]+)?(/wp-.*) $2 last;
+  rewrite ^(/[^/]+)?(/.*\.php) $2 last;
+}
+```
+
+This config will allow nginx to play nice with your subdirectory-based multisite network.
+
+Save the files and re-run `./bin/start`.
```

Comment:
You actually need to run `docker-compose build` before you run `./bin/start` so that the updates to the nginx config file make it into the Docker image, but this causes another problem -- it reverts your edits to the `wp-config.php` in the container.

We can avoid the need for `docker-compose build` if we mount `conf` as a volume in the container at `/var/www/html/conf` in `docker-compose.yml`.

Reply:
I believe that you'd also need to run `docker-compose build` to enforce your updates to the `scripts/setup.sh` script, including all of the multisite constants.

Assuming that you update both your `scripts/setup.sh` and `conf/nginx/nginx-site.conf` files beforehand, it should be ok that running `docker-compose build` will revert the earlier changes to the `wp-config.php`. Assuming we're following these instructions, the only thing we would have added to `wp-config.php` before modifying these files would be adding the `WP_ALLOW_MULTISITE` constant, which can be removed after multisite has been installed.

Given the mostly static nature of the config files for the docker container, I think it feels better to have to run the docker build command manually, rather than mounting both the `conf` and `scripts` volumes via `docker-compose.yml`. Especially if we plan to work a multisite option into a future installation script.